api/src/main/java/org/apache/iceberg/types/CheckCompatibility.java

```diff
@@ -53,7 +67,24 @@ public static List<String> writeCompatibilityErrors(Schema readSchema, Schema wr
    */
   public static List<String> writeCompatibilityErrors(
       Schema readSchema, Schema writeSchema, boolean checkOrdering) {
-    return TypeUtil.visit(readSchema, new CheckCompatibility(writeSchema, checkOrdering, true));
+    return writeCompatibilityErrors(readSchema, writeSchema, checkOrdering, 2);
```

Comment:
same comment on commenting literal (or maybe there are already defined constants someplace in the code base)?